### ISSUE TITLE
Redirection Issue: BitPay redirects to "example.com"

### ISSUE THREAD

**Issue by VikBlockEater** [bitpay/php-bitpay-client-v2#114]
> Are you using dev-master or a tagged version?

Its v6.0x.

> Which version of PHP are you on?

8.0 on staging and 8.1 on local (but issue is same on both versions).

>Are you using the SDK directly, or are any other frameworks / platforms involved? I noticed you mentioned livewire, is that [this framework](https://laravel-livewire.com/), specifically?

Yes its Laravel Livewire.

>There may be a typo in this part here, where it should be $this->balanceAmount where you have $this>balanceAmount

Yes I realised it earlier and rectified that but issue was still there.

>I believe what you may want to consider is creating an Invoice object, setting its values, and then passing that entire object into the createInvoice function.

As per your suggestion I refactored my code as below

`// creating BitPay client. Client instantiated successfully
$bitpay = BitPaySDK\Client::create()->withFile(base_path() . '/config/bitpay.json');

// creating invoice object
$invoice = new \BitPaySDK\Model\Invoice\Invoice($this->balanceAmount,  \BitPaySDK\Model\Currency::USD);

// setting redirect url on invoice
$invoice->setRedirectURL("https://myapp.com/");

// setting redirect url on closing the invoice
$invoice->setCloseURL("https://myapp.com/");

// setting auto redirect to true
$invoice->setAutoRedirect(true);

// creating invoice
$bitpay->createInvoice($invoice);

// the url of the invoice to be redirected to BitPay
$invoice_redirect_url = $invoice->getURL();

// upon click on livewire component redirect to invoice BitPay URL
return redirect($invoice_redirect_url);
`
After implementing the above change, I observed difference in the Invoice object. Detailed after vs before comparison could be found here [https://docs.google.com/spreadsheets/d/1Vxnuh3RTPuSY4EBMSOlufPG0wPv661qetab_oM6QHGs/edit?usp=sharing](url)

Conclusively new code (along with other differences in response) is not generating the invoice URL.

If you need some other clarification then feel free to communicate back.

@bobbrodie Thank you very much your supportive response.

_Originally posted by @VikBlockEater in https://github.com/bitpay/php-bitpay-client-v2/issues/103#issuecomment-1270102887_

**Comment by bobbrodie** [comment]
Hi @VikBlockEater we've recently released version 7.0.0 of the plugin which addresses many issues from the 6.x version, and I've posted example code in #111.

I have inspected the array output of an invoice, and do see that the array of the Invoice does have `notificationURL`. We are ensuring that it is sent to the [API](https://bitpay.com/api/#rest-api-resources-invoices-create-an-invoice), and if it is then we will check with the API team.

**Comment by VikBlockEater** [comment]
Hello @bobbrodie ,

As per your suggestion I have updated the SDK

<img width="1656" alt="image" src="https://user-images.githubusercontent.com/110638687/195777204-e6fcdfa8-b503-4e83-b6d9-28698313bab2.png">

The problem is still the same. Let me precisely explain again.

When I create invoice using the below code

<img width="878" alt="image" src="https://user-images.githubusercontent.com/110638687/195778409-a4bee721-1ba6-44e5-8d28-431784f91f1c.png">

`$invoice_redirect_url` contains the value. In invoice object it's key is `_url`

<img width="661" alt="image" src="https://user-images.githubusercontent.com/110638687/195783822-b3946263-acf5-45bf-8fb0-cfabe782aaf6.png">

But when I create the invoice using below method as suggested by you and I also feel it more sensible

<img width="894" alt="image" src="https://user-images.githubusercontent.com/110638687/195782783-fff85db1-22d6-4d1c-8b01-d8991afef95a.png">

`$invoice_redirect_url` contains null. In invoice object it's key is `_url`

<img width="764" alt="image" src="https://user-images.githubusercontent.com/110638687/195785005-22ef484e-3d93-4687-9a1c-e17864108a23.png">

Detailed comparison of Invoice object in both scenarios could be found here https://docs.google.com/spreadsheets/d/1Vxnuh3RTPuSY4EBMSOlufPG0wPv661qetab_oM6QHGs/edit?usp=sharing

I have included the images instead of code because code is already provided in the issue thread.

In case of any clarification please feel free to ask for more clarity.

**Comment by bobbrodie** [comment]
@VikBlockEater we are aware of the issue with the redirect URL, and the SDK is submitting it as expected by the API. We are working with the API team to address.

**Comment by VikBlockEater** [comment]
@bobbrodie thanks for your prompt and quick response.

Issue#1:
My first issue was that after redirection for invoice payment to test.bitpay.com, whenever I was closing the invoice payment interface, it was redirecting me to "example.com".

Reason:
I was mistakenly setting the values to the `invoice` object after creating the invoice using `createInvoice()` method. As you said, in the backend these values were not updated in bitpay system.

Issue#2:
I was not getting the invoice url when I changed the approach of instantiating invoice object.

Reason:
I was calling `getUrl()` method on Invoice object not on the object resulted by `createInvoice()` method.

Solution:
As you suggested, I created the invoice object first `$invoice = new Invoice()`, then set the values to that object, then create invoice passing the invoice object as argument `$bitpay_invoice = $client->createInvoice($invoice)`. Afterwards I called the `getUrl()` method on `$bitpay_invoice`

Result:
Invoice object properties are set accurately. Now I get the invoice url and from the invoice payment interface I am successfully redirected to myapp.

@bobbrodie Once again Thank You so much for consistent support. I am closing this issue.
